packages/alfa-style/src/property/z-index.ts

```diff
@@ -0,0 +1,24 @@
+import { Integer, Keyword } from "@siteimprove/alfa-css";
+import { Parser } from "@siteimprove/alfa-parser";
+
+import { Longhand } from "../longhand.js";
+
+const { either } = Parser;
+
+const parse = either(Keyword.parse("auto"), Integer.parse);
+
+type Specified = Keyword<"auto"> | Integer;
+type Computed = Keyword<"auto"> | Integer.Canonical;
+
+/**
+ * {@link https://developer.mozilla.org/en-US/docs/Web/CSS/z-index}
+ * @internal
+ */
+export default Longhand.of<Specified, Computed>(
+  Keyword.of("auto"),
+  parse,
+  (value) => value.map((zIndex) => zIndex.resolve()),
```

Comment:
**Note:** This should be good, but it might break in the weird case of a length division.
```suggestion
  (value, style) => value.map((zIndex) => zIndex.resolve(Resolver.length(style))),
```
would be the safe bet.

Reply:
Just out of curiosity, is a length division when there units involved? Units should be forbidden in z-index values and that seems to be true even if they cancel out; I tested `{ z-index: "calc(1px / 1px)"; }` on https://jigsaw.w3.org/css-validator/validator and it gave a parsing error and in chrome the computed value becomes `auto`.

Reply:
It also gave a parsing error when the syntax was valid 😅; `.foo { z-index: calc(1px / 1px); }`

Reply:
We probably actually need a resolver that _doesn't_ cancel units in this case to conform to the spec 🙈 But it might be too much of a corner case 🤷🏼

Reply:
Based on [Type checking](https://drafts.csswg.org/css-values/#calc-type-checking) and [Operations on types](https://drafts.css-houdini.org/css-typed-om-1/#cssnumericvalue-multiply-two-types), I think that these do technically "match" the `<number>`/`<integer>` types. And since, according to [Math expressions](https://drafts.csswg.org/css-values/#math):

> A [math function](https://drafts.csswg.org/css-values/#math-function) represents a numeric value, one of (…)[<number>](https://drafts.csswg.org/css-values/#number-value), [<integer>](https://drafts.csswg.org/css-values/#integer-value) (…), and can be used wherever such a value would be valid.

I understand the specs as not forbidding it…
I totally get why browsers think it's madness and refuse this, however.